File: maestro-sdk/android/src/main/java/dev/mobile/maestro/sdk/MaestroSdk.kt
package dev.mobile.maestro.sdk

import dev.mobile.maestro.sdk.mockserver.MaestroMockServerSdk

object MaestroSdk {

    internal var projectId: String? = null

    @JvmStatic
    fun init(projectId: String) {
        this.projectId = projectId
    }

    fun mockServer(): MaestroMockServerSdk {
        ensureInitialised()

        return MaestroMockServerSdk()
    }

    private fun ensureInitialised() {
        if (projectId == null) {
            error("Maestro SDK is not initialised. Please call MaestroSdk.init(projectId) first.")
        }
    }

}

File: maestro-sdk/ios/Maestro SDK/Maestro SDK/MaestroSDK.swift
import Foundation

public struct MaestroSdk {
    
    @available(*, unavailable) private init() {}
    
    static var projectId: String? = nil
    
    public static func setup(projectId: String) {
        MaestroSdk.projectId = projectId
    }
    
    public static func mockServer() -> MaestroMockServerSdk {
        return MaestroMockServerSdk()
    }
    
}
